src/tools/pub/cmd/pub-token.md

```diff
@@ -0,0 +1,91 @@
+---
+title: dart pub token
+description: Manage authentication tokens for package repositories.
+---
+
+_pub_ _token_ is one of the subcommands of the [pub command](/tools/pub/cmd).
+
+It is used to manage a store of secret tokens for authenticating
+against third-party servers when [publishing](pub-lish) packages and
+[retrieving](pub-get) dependencies.
+
+The tokens are stored in a
+[user-wide config dir](https://github.com/dart-lang/cli_util/blob/71ba36e2554f7b7717f3f12b5ddd33751a4e3ddd/lib/cli_util.dart#L88-L118). 
+
+It has three subcommands: `add`, `list` and `remove`.
+
+If you try to `dart pub get` and have a [dependency](/tools/pub/dependencies) hosted
+on a private repository you _may_ be asked to provide credentials:
+
+```terminal
+$ dart pub get
+Resolving dependencies... 
+https://some-package-repo.com/my-org/my-repo package repository requested authentication! You can provide credential using:
+    pub token add https://some-package-repo.com/my-org/my-repo
+
+Go to https://some-package-repo.com and log in to obtain your token. 
+```
+
+The last line is a message the server can provide to help you obtaining a token.
+Some servers might not provide such a message.
+
+## Adding credentials `dart pub token add`
+
+To enter the credentials use `dart pub token add`, 
+and type the credential on stdin.
+
+```terminal
+$ dart pub token add https://some-package-repo.com/my-org/my-repo
+Enter secret token: <Type token on stdin>
+ Requests to "https://some-package-repo.com/my-org/my-repo" will now be 
+ authenticated using the secret token.
+```
+
+{{site.alert.note}}
+  The token is input on stdin rather than as a command line option to avoid it
+  ending up in the shell history such as `~/.bash_history`.
+{{site.alert.end}}
+
+In a scripting situation you can store the secret in an environment variable and
+ use `dart pub token add --env-var`.
+
+```terminal
+$ dart pub token add https://other-package-repo.com/ --env-var TOKEN_VAR
+Requests to "https://other-package-repo.com/" will now be authenticated using the secret token stored in the environment variable "TOKEN_VAR".
```

Comment:
```suggestion
```terminal
export TOKEN_VAR='<secret token>'
$ dart pub token add https://other-package-repo.com/ --env-var TOKEN_VAR
Requests to "https://other-package-repo.com/" will now be authenticated using the secret token stored in the environment variable "TOKEN_VAR".
```

Reply:
I instead explain how to use the secrets functionality of known CI systems.